pkg/logs/input/docker/launcher.go

```diff
@@ -298,10 +299,15 @@ func (l *Launcher) getFileSource(container *Container, source *config.LogSource)
 }
 
 // getPath returns the file path of the container log to tail.
-// The pattern looks like /var/lib/docker/containers/{container-id}/{container-id}-json.log
-func (l *Launcher) getPath(id string) string {
-	filename := fmt.Sprintf("%s-json.log", id)
-	return filepath.Join(basePath, id, filename)
+func getPath(id string) string {
+	var tpl string
+	switch coreConfig.Datadog.GetBool("logs_config.use_podman_logs") {
+	case false:
+		tpl = filepath.Join(basePath, "$ID/$ID-json.log")
+	case true:
+		tpl = "/var/lib/containers/storage/overlay-containers/$ID/userdata/ctr.log"
+	}
+	return strings.ReplaceAll(tpl, "$ID", id)
```

Comment:
nit: Not a blocker but I think this helper would be easier to maintain in the future with an early exit and `fmt.Sprintfs`. Something like:

```go
if coreConfig.Datadog.GetBool("logs_config.use_podman_logs") {
    return fmt.Sprintf("/var/lib/containers/storage/overlay-containers/%s/userdata/ctr.log", id)
}

return filepath.Join(basePath, id, fmt.Sprintf("%s-json.log", id))
```